File: Classes/EVEOnlineAPI/Corporation/EVEOutpostServiceDetail.m
//
//  EVEOutpostServiceDetail.m
//  EVEOnlineAPI
//
//  Created by Artem Shimanski on 1/7/11.
//  Copyright 2011 __MyCompanyName__. All rights reserved.
//

#import "EVEOutpostServiceDetail.h"


@implementation EVEOutpostServiceDetailItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"stationID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"ownerID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"serviceName":@{@"type":@(EVEXMLSchemePropertyTypeString)},
				   @"minStanding":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"surchargePerBadStanding":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"discountPerGoodStanding":@{@"type":@(EVEXMLSchemePropertyTypeScalar)}};
	
	return scheme;
}

@end


@implementation EVEOutpostServiceDetail

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"outpostServiceDetails":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEOutpostServiceDetailItem class]}};
	
	return scheme;
}

@end

File: EVEAPI/EVEAPI/EVEOutpostServiceDetail.swift
//
//  EVEOutpostServiceDetail.swift
//  EVEAPI
//
//  Created by Artem Shimanski on 30.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit

public class EVEOutpostServiceDetailItem: EVEObject {
	public var stationID: Int = 0
	public var ownerID: Int = 0
	public var serviceName: String = ""
	public var minStanding: Double = 0
	public var surchargePerBadStanding: Double = 0
	public var discountPerGoodStanding: Double = 0
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"stationID":EVESchemeElementType.Int(elementName:nil, transformer:nil),
			"ownerID":EVESchemeElementType.Int(elementName:nil, transformer:nil),
			"serviceName":EVESchemeElementType.String(elementName:nil, transformer:nil),
			"minStanding":EVESchemeElementType.Double(elementName:nil, transformer:nil),
			"surchargePerBadStanding":EVESchemeElementType.Double(elementName:nil, transformer:nil),
			"discountPerGoodStanding":EVESchemeElementType.Double(elementName:nil, transformer:nil),
		]
	}
}


public class EVEOutpostServiceDetail: EVEResult {
	public var outpostServiceDetails: [EVEOutpostServiceDetailItem] = []
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"outpostServiceDetails":EVESchemeElementType.Rowset(elementName: nil, type: EVEOutpostServiceDetailItem.self, transformer: nil),
		]
	}
}
